### PR TITLE
Fixes #80 - Refactor ChartTestStarter

### LINKED ISSUE #80: Refactor ChartTestStarter
* Easier to understand and maintain
* Showcase performance issues
* Removes all notify properties changes event from test code
* Only one view 
* Use DI instead of new everything
* Add meaningful buttons on side bar
  - Add N observed Data (where N input in control)
  - Add N curves (where N input in control)
  - Prevent view update (either button or check box)
  - Refresh all

### PR DESCRIPTION
**Pull request by rwmcintosh**
Keeping some of the old functionality, but added the required stuff.

Stop/Start chart updates has an implementation, but does not do what is expected yet. I am calling BeginUpdate/EndUpdate in these cases, but that does not allow a nice clean 'prevent drawing' and then 'draw everything'. I expect that will be hooked in as it's implemented. Not sure we have this now and I did not touch the actual chart code for this refactor.

### CODE REVIEW

File: tests/OSPSuite.Starter/Presenters/ChartTestPresenter.cs [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+
+         ChartDisplayPresenter.DataSource = Chart;
+         Chart.ChartSettings.DiagramBackColor = Color.White;
+         ChartDisplayPresenter.DragOver += onChartDisplayDragOver;
+         ChartEditorPresenter.DragOver += onChartDisplayDragOver;
+
+         ReloadMenus();
+      }
+
+      public void InitializeRepositoriesWithOriginalData()
```
Comment (rwmcintosh) [inline tests/OSPSuite.Starter/Presenters/ChartTestPresenter.cs]: That gives the original data as it was before refactor

File: tests/OSPSuite.Starter/Presenters/ChartTestPresenter.cs [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+         ChartEditorPresenter.AddDataRepositories(newRepositories);
+         addNewRepositories(newRepositories);
+      }
+
+      private void addNewRepositories(IEnumerable<DataRepository> newRepositories)
+      {
+         _dataRepositories.AddRange(newRepositories);
+      }
+
+      public void AddObservations(int numberOfObservations)
```
Comment (rwmcintosh) [inline tests/OSPSuite.Starter/Presenters/ChartTestPresenter.cs]: Add N observations

File: tests/OSPSuite.Starter/Presenters/ChartTestPresenter.cs [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+         addNewCurvesToChart(newRepositories);
+         ChartEditorPresenter.SelectDataColumns();
+      }
+
+      private void addNewCurvesToChart(List<DataRepository> newRepositories)
+      {
+         newRepositories.Each(repository => repository.AllButBaseGrid().Each(column => ChartEditorPresenter.AddCurveForColumn(column.Id)));
+      }
+
+      public void AddCalculations(int numberOfCalculations)
```
Comment (rwmcintosh) [inline tests/OSPSuite.Starter/Presenters/ChartTestPresenter.cs]: Add N calculations

**Comment by msevestre** [comment]
There are some conflicts. Can you rebase of version7.2.0

**Comment by msevestre** [comment]
I would remove the start stop implementation because it's not what we will be using. Mark is as not implemented for now

File: tests/OSPSuite.Starter/Presenters/ChartTestPresenter.cs [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+      }
+
+      public IChartEditorPresenter ChartEditorPresenter => _chartEditorAndDisplayPresenter.EditorPresenter;
+
+      private static Func<DataColumn, PathElements> setQuantityPathDefinitions(IContainer model, IDataColumnToPathElementsMapper dataColumnToPathColumnValuesMapper)
+      {
+         return col => dataColumnToPathColumnValuesMapper.MapFrom(col, model.RootContainer);
+      }
+
+      private static bool showDataColumnInDataBrowserDefinition(DataColumn dataColumn)
```
Comment (msevestre) [inline tests/OSPSuite.Starter/Presenters/ChartTestPresenter.cs]: ??

**Pull request merged**